### PR TITLE
DEVEXP-433: Report image pruner's install status in Prometheus

### PR DESCRIPTION
**Pull request by adambkaplan**
* Add a new Gauge metric which reports the installation status of the
automatic impage pruner.
* Alert if the image pruner is not installed and enabled
* TEMP: report that the image pruner has been installed for CI to pass

### CODE REVIEW

**Comment by adambkaplan** [comment]
@openshift/openshift-team-monitoring PTAL. We want to send the `install_info` metric to Telemeter.

File: manifests/09-prometheus-rules.yaml [hunk 1]
```diff
@@ -28,3 +28,18 @@ spec:
           DeploymentConfigs which reference ImageStreamTags may not work as
           expected. Please configure storage and update the config to Managed
           state by editing configs.imageregistry.operator.openshift.io.
+  - name: ImagePruner
+    rules:
+    - alert: ImagePruningDisabled
+      expr: image_registry_operator_image_pruner_install_info{} < 2
```
Comment (metalmatze) [inline manifests/09-prometheus-rules.yaml]: You don't need the empty `{}`. 
```suggestion
      expr: image_registry_operator_image_pruner_install_info < 2
```

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -10,10 +10,17 @@ var (
 		Name:      "reconfigured_total",
 		Help:      "Total times the image registry's storage was reconfigured.",
 	})
+	imagePrunerInfo = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "image_registry_operator",
+		Subsystem: "image_pruner",
+		Name:      "install_info",
```
Comment (metalmatze) [inline pkg/metrics/metrics.go]: We tend to simply do the entire thing in name these days. I think the other fields might even be deprecated going forward.
`Name: image_registry_operator_image_pruner_install_info`.

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -10,10 +10,17 @@ var (
...
+		Help:      "Installation status code related to the automatic image pruning feature. 0 = not installed, 1 = suspended, 2 = enabled",
+	}, []string{})
```
Comment (brancz) [inline pkg/metrics/metrics.go]: There is no need for this to be a GaugeVec if there are no labels, and the `WithLabelValues` call is also not needed in that case.

**Comment by brancz** [comment]
Judging by the code I see, this alert will always fire as we always set the metric to not enabled and not installed. I assume this is currently just done to ensure that this actually fails e2e tests when set that way?

File: manifests/09-prometheus-rules.yaml [hunk 4]
```diff
@@ -28,3 +28,18 @@ spec:
...
+      labels:
+        severity: warning
+      annotations:
+        message: |
+          Automatic image pruning is not enabled. Regular pruning of images
+          no longer referenced by ImageStreams is strongly recommened to ensure
```
Comment (simonpasquier) [inline manifests/09-prometheus-rules.yaml]: `recommened` -> `recommended`

**Comment by adambkaplan** [comment]
@brancz the current code that sets the metric & fires the alert is purely for test purposes. On new clusters the image pruner will be installed and we shouldn't get the alert.

Upgraded clusters are the more interesting case - they won't automatically get the pruner. In this case we _want_ the alert fired so admins are prompted to install it afterwards.

**Comment by brancz** [comment]
Ack, that makes sense. In that case once this is ready from your side I'm happy with it.

File: manifests/09-prometheus-rules.yaml [hunk 5]
```diff
@@ -28,3 +28,18 @@ spec:
           DeploymentConfigs which reference ImageStreamTags may not work as
           expected. Please configure storage and update the config to Managed
           state by editing configs.imageregistry.operator.openshift.io.
+  - name: ImagePruner
+    rules:
+    - alert: ImagePruningDisabled
+      expr: image_registry_operator_image_pruner_install_info < 2
```
Comment (adambkaplan) [inline manifests/09-prometheus-rules.yaml]: @brancz @metalmatze @simonpasquier perhaps `install_status` is a better metric name? This metric sits in between "metadata" info and a real value.

Comment (brancz) [reply manifests/09-prometheus-rules.yaml]: Yeah that works as well. It’s true that info is really meant for constant value time series.

**Comment by adambkaplan** [comment]
/assign @coreydaley 

We can land this as is to stage our alert that pruning is not installed and/or enabled.

**Comment by adambkaplan** [comment]
/retest

AWS rate limit exceeded

**Comment by adambkaplan** [comment]
/assign @dmage 

/cc @coreydaley 

Tests are green

**Comment by adambkaplan** [comment]
/hold

Upgrade tests are getting bit by https://bugzilla.redhat.com/show_bug.cgi?id=1793051

**Pull request merged**